pkg/volume/portworx/portworx_util.go

```diff
@@ -41,12 +42,17 @@ type PortworxVolumeUtil struct {
 
 // CreateVolume creates a Portworx volume.
 func (util *PortworxVolumeUtil) CreateVolume(p *portworxVolumeProvisioner) (string, int, map[string]string, error) {
-	hostname := p.plugin.host.GetHostName()
-	client, err := util.osdClient(hostname)
-	if err != nil {
-		return "", 0, nil, err
+	if util.portworxClient == nil || !isValid(util.portworxClient) {
```

Comment:
This `util.portworxClient` and `driver` discovery is copied to many places, can you make a function out of it?
```
driver, err := util.getPortworxDriver(p.plugin.host)
if err != nil ...
```